### DISCUSSION TITLE
请教下V3支付回调时验签是在哪调用的

### DISCUSSION THREAD

**Discussion by a8ss** [w7corp/easywechat#2509] [General]
看到了验签的方法

> src/Pay/ResponseValidator.php

但未找到调用地方。

**Comment by sunkangchina** [comment]
V3手册里面有吧 https://easywechat.com/6.x/pay/server.html#%E5%86%85%E7%BD%AE%E4%BA%8B%E4%BB%B6%E5%A4%84%E7%90%86%E5%99%A8
~~~
$server->handlePaid(function (Message $message, \Closure $next) {
    // $message->out_trade_no 获取商户订单号
    // $message->payer['openid'] 获取支付者 openid
    // 🚨🚨🚨 注意：推送信息不一定靠谱哈，请务必验证
    // 建议是拿订单号调用微信支付查询接口，以查询到的订单状态为准
    return $next($message);
});

~~~

**Comment by overtrue** [comment]
自己 new 来调用哈，文档后续补上： https://github.com/w7corp/easywechat/blob/6.x/src/Pay/ResponseValidator.php

```php
$validator = new ResponseValidator($app->gerMerchant());

$response = $app->getClient()->get(...);

try {
    $validator->validate($response);
} catch(\Throwable $e) {
     // 验证失败
}
```

**Reply by sunkangchina** [reply]
> 自己 new 来调用哈，文档后续补上： https://github.com/w7corp/easywechat/blob/6.x/src/Pay/ResponseValidator.php
> 
> ```
> $validator = new ResponseValidator($app->gerMerchant());
> 
> $response = $app->getClient()->get(...);
> 
> try {
>     $validator->validate($response);
> } catch(\Throwable $e) {
>      // 验证失败
> }
> ```

$server->handlePaid 这个没有自动验证？

**Reply by overtrue** [reply]
没有，不是所有人都要求验证

**Reply by cnxfire** [reply]
发来的本来就是解密过的吧，不验签应该没啥事
